flink-baseline/flink-canary/src/main/java/com/ness/flink/canary/pipeline/function/HealthCheckFunction.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021-2023 Ness Digital Engineering
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ness.flink.canary.pipeline.function;
+
+import com.ness.flink.canary.pipeline.domain.KafkaConfigs;
+import java.util.Collection;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.streaming.api.functions.ProcessFunction;
+import org.apache.flink.util.Collector;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DescribeClusterOptions;
+import org.apache.kafka.common.Node;
+
+@Slf4j
+public class HealthCheckFunction extends ProcessFunction<KafkaConfigs, String>{
+    private static final long serialVersionUID = 1L;
+    @Override
+    public void processElement(KafkaConfigs value, ProcessFunction<KafkaConfigs, String>.Context ctx, Collector<String> out) throws Exception {
+        String result;
+
+        Properties props = new Properties();
+        String bootStrapServers = value.getBootStrapServers();
+        props.put("bootstrap.servers", bootStrapServers);
```

Comment:
I don't think that it will work on Confluent Cloud because I don't see required options for it.